gpflow/optimizers/scipy.py

```diff
@@ -97,20 +114,34 @@ def initial_parameters(cls, variables: Sequence[tf.Variable]) -> tf.Tensor:
 
     @classmethod
     def eval_func(
-        cls, closure: LossClosure, variables: Sequence[tf.Variable], compile: bool = True
+        cls,
+        closure: LossClosure,
+        variables: Sequence[tf.Variable],
+        compile: bool = True,
+        unconnected_gradients: str = tf.UnconnectedGradients.ZERO,
```

Comment:
Are we sure this is the behaviour we want? Maybe it would be safer to default to `None`, so that there is an obvious failure if the unconnected gradient is indeed a bug? And users can pick `ZERO` if they're ok with a bit more risk?